includes/gateways/class-wpsms-gateway-_textplode.php

```diff
@@ -4,133 +4,134 @@
 
-        // Enable api key
-        $this->has_key = true;
+        // Optional: set defaults or dynamic help
+        $this->help = $this->help ?: esc_html__('Configure Textplode API Key and From Name.', 'wp-sms');
     }
 
     public function SendSMS()
     {
-        /**
-         * Modify sender number
-         *
-         * @param string $this ->from sender number.
-         * @since 3.4
-         *
-         */
+        // Let WP SMS apply number cleaning / country code if enabled in settings
         $this->from = apply_filters('wp_sms_from', $this->from);
-
-        /**
-         * Modify Receiver number
-         *
-         * @param array $this ->to receiver number
-         * @since 3.4
-         *
-         */
-        $this->to = apply_filters('wp_sms_to', $this->to);
-
-        if (!class_exists('Textplode')) {
-            $this->log($this->from, $this->msg, $this->to, __('The Textplode class could not be found. Please ensure the Textplode library or plugin is properly loaded.', 'wp-sms'), 'error');
-
-            return new \WP_Error(
-                'textplode-missing',
-                __('The Textplode class could not be found. Please ensure the Textplode library or plugin is properly loaded.', 'wp-sms')
-            );
-        }
-
-        // Get the credit.
-        $credit = $this->GetCredit();
-
-        // Check gateway credit
-        if (is_wp_error($credit)) {
-            // Log the result
-            $this->log($this->from, $this->msg, $this->to, $credit->get_error_message(), 'error');
-
-            return $credit;
-        }
-
-        /**
-         * Modify text message
-         *
-         * @param string $this ->msg text message.
-         * @since 3.4
-         *
-         */
-        $this->msg = apply_filters('wp_sms_msg', $this->msg);
-
-        // Init class
-        $textplode = new \Textplode($this->has_key);
-
-        // Add recipient
-        foreach ($this->to as $to) {
-            $textplode->messages->add_recipient($to, array());
-        }
-
-        // Set From Name
-        $textplode->messages->set_from($this->from);
-
-        // Set Message
-        $textplode->messages->set_message($this->msg);
-
-        // Send sms
-        $result = $textplode->messages->send();
-
-        // Check result
-        if (!$result) {
-            // Log the result
-            $this->log($this->from, $this->msg, $this->to, $result, 'error');
-
-            return new \WP_Error('send-sms', $result);
+        $this->to   = apply_filters('wp_sms_to',   $this->to);
+        $this->msg  = apply_filters('wp_sms_msg',  $this->msg);
+
+        try {
+            // Build recipients array for Textplode API format
+            $recipients = [];
+            foreach ((array) $this->to as $number) {
+                $recipients[] = [
+                    'phone_number' => $number,
+                    'merge'        => []
+                ];
+            }
```

Comment:
Suggestion: Using `array_map` here would be more appropriate and maintain a cleaner approach.